pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -134,6 +136,32 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enable
 		return nil, fmt.Errorf("cannot parse node port: %v", err)
 	}
 
+	pluginName := options.NetworkConfig.NetworkPluginName
+	if sdnovs.IsOpenShiftNetworkPlugin(pluginName) {
+		// Detect any plugin mismatches between node and master
+		clusterNetwork, err := originClient.ClusterNetwork().Get(sdnapi.ClusterNetworkDefault)
+		if kerrs.IsNotFound(err) {
+			return nil, fmt.Errorf("detected network plugin mismatch between OpenShift node(%q) and master(non OpenShift network plugin)", pluginName)
+		} else if err != nil {
+			return nil, fmt.Errorf("cannot fetch %q cluster network: %v", sdnapi.ClusterNetworkDefault, err)
+		}
+
+		if clusterNetwork.PluginName != strings.ToLower(pluginName) {
+			if clusterNetwork.PluginName == "" {
+				// Do not return error in this case
+				glog.Warningf("either there is network plugin mismatch between OpenShift node(%q) and master or OpenShift master is running an older version where we did not persist plugin name", pluginName)
```

Comment:
This is user facing, so should follow the convention of other warning messages in this file.